src/Documents/Document.ts

```diff
@@ -0,0 +1,15 @@
+import {IDocument} from './IDocument';
+
+export class Document implements IDocument {
+  public entityName(): string {
+      return this.constructor.name;
+  }
+
```

Comment:
how i should serialize them ?
your Pythin client accepts any objects, it also support linked documents where the property is a another class

also, a lot of mongo clients has own models classes with validation, relations etc
i would use IDocument/Document based on Backbone Model